### DISCUSSION TITLE
TumblThree skips a lot of pictures when using the file rename functionality

### DISCUSSION THREAD

**Discussion by vickileekx** [TumblThreeApp/TumblThree#336] [Q&A (Question & Answer)]
Hello, I love this software, i wanna thank the devs for their dedication and hard work after all these years, i've tried all of the tumblr downloaders available on the internet and this one is the best,  I often use it to back up my personal blog, but there is something I still don't understand, for example when i download a blog, why are the pictures in the folder not in the same order they were reblogged/posted on the blog? 

I've recently tried to use the file rename functionality hoping it would download the images in the same order they were reblogged but strangely when I use this feature the software does not download the whole blog, it skips a lot of posts and will only download a small portion. I have 2900 pictures on my blog and TumblThree only downloads 315 pictures if i use the file rename functionality,  I used the template "%d-%u-%i-%f" . Maybe i am doing something wrong?

However when i use the default %f template it will download the whole blogs with all the pictures etc, it's really weird. 

Sorry if i'm a bit bad at expressing my thoughts , english isn't my first language i've tried my best to explain my issue.

**Comment by thomas694** [comment]
Hello,
thanks for your compliments. It's nice to hear this one works best for you.
There's no need to say sorry. Let's try to answer your questions.

Tumblr doesn't create filenames that are in chronological order. For example the first blog entry gets filename "6e84e8ca0d66329b98554a2d203d411396e7fc7a.jpg", the second "1d27d86200641a18e0176e1db61f711c864d0145.jpg" and so on. Even Tumblr's old style filenames (tumblr_nyn0mm4XkT1qh1x6ko2_1280.jpg) that you can still find in some blogs, are not chronological. If you sort them alphabetically in windows explorer, you get a total different order. Even if you sort them by date in windows explorer, you won't get the chronological order, because our application downloads the files in parallel, so the order is not determined.
That's one of the reasons we introduced the file rename functionality.

Your second question. Yes, indeed, you've overseen something. Your second test was "continuing" after some files have already been downloaded.
If you want to do a comparison with different filename templates for a blog "dogs.tumblr.com", you should always close TumblThree, rename the folder "dogs" and adjust the blog's files "dogs.tumblr" ("LastId": 0) and "dogs_files.tumblr" ("Entries": [ ]) inside the index folder. Either edit these files or just delete them and make all blog settings in the application like before (except different filename template).

Now compare the two folders. Then you'll see that it actually works. And that either of %d, %u or %i, depending on your needs, would be enough to sort them chronologically.

Feel free to ask more questions, if something is unclear.

**Comment by vickileekx** [comment]
Thanks for your answer and taking to time to help me, I followed your advice and tried other template combinations this morning, I finally managed to get the files to be in the same chronological order as they were posted on the blog by using the template "%i-%x" and sorting/filtering the files by "date downloaded" in the windows explorer folder. Now it's exactly the way i wanted it to be! Thank you :)
This functionality works great.

**Comment by DReaper** [comment]
I'm a little confused.

I'm having a similar issue, where it pulls all files with the default %f. I changed it to %d-%f, so that it includes the post date.

For some reason, it pulls hundreds less, pulling up to 2018 when there's posts up to 2021. I renamed the %f files as a reference, and made sure the index folder was cleared before trying with %d-%f.

**Comment by thomas694** [comment]
Yes, there's an error when using the %d token. It's only visible with some blogs and Tumblr (tag) searches.
Anyway, it will be solved in the next version.

**Comment by thomas694** [comment]
New version 1.6.0 available.
